src/framework/components/particle-system/component.js

```diff
@@ -108,6 +109,7 @@ var depthLayer;
  * @property {boolean} depthWrite If enabled, the particles will write to the depth buffer. If disabled, the depth buffer is left unchanged and particles will be guaranteed to overwrite one another in the order in which they are rendered.
  * @property {boolean} noFog Disable fogging.
  * @property {boolean} localSpace Binds particles to emitter transformation rather then world space.
+ * @property {boolean} screenSpace Renders particles in 2D screen space. This needs to be set when particle system is part of hierarchy with {@link pc.ScreenComponent} at its root, and allows particle system to integrate with the rendering of {@link pc.ElementComponent}s. Note that an entity with ParticleSystem component cannot be parented directly to {@link pc.ScreenComponent}, but has to be a child of a {@link pc.ElementComponent}, for example {@link pc.LayoutGroupComponent}.
```

Comment:
The screen doesn't have to be at the root, right? It just has to be an ancestor in the hierarchy, right?

Reply:
Also, what's the motivation for having the `screenSpace` property anyway? If the particlesystem component is a descendant of an entity with a screen component, shouldn't screen space be assumed automatically?

Reply:
I changed the doc to use "ancestor", thanks.

Reply:
> Also, what's the motivation for having the `screenSpace` property anyway? If the particlesystem component is a descendant of an entity with a screen component, shouldn't screen space be assumed automatically?

The motivation is avoiding the complexity / performance impact of that. If a child entity is added to a parent, the code would need to check if any of the descendant entities contain ParticleSystem and if there is a Screen component ascendant. setParent could get expensive.